Panic observed when flushing the pyroscope client

```
2025-07-08 16:32:18.279	
panic: sync: WaitGroup is reused before previous Wait has returned
goroutine 368 [running]:
sync.(*WaitGroup).Wait(0xc0173d8001?)
	sync/waitgroup.go:120 +0x74
github.com/grafana/pyroscope-go/upstream/remote.(*Remote).Flush(0xc00316b860?)
	github.com/grafana/pyroscope-go@v1.2.0/upstream/remote/remote.go:138 +0x25
github.com/grafana/pyroscope-go.(*Session).takeSnapshots(0xc00316b860)
	github.com/grafana/pyroscope-go@v1.2.0/session.go:159 +0x21a
github.com/grafana/pyroscope-go.(*Session).Start.func1()
	github.com/grafana/pyroscope-go@v1.2.0/session.go:184 +0x49
created by github.com/grafana/pyroscope-go.(*Session).Start in goroutine 1
	github.com/grafana/pyroscope-go@v1.2.0/session.go:182 +0x96
```

This appears to happen if there are concurrent attempts to Upload and Flush. 

https://github.com/grafana/pyroscope-go/blob/03917688830b772383d06acf96685fc8a1d51eff/upstream/remote/remote.go#L124-L139

--- a/go.mod
+++ b/go.mod
@@ -139,7 +139,7 @@ require (
 	github.com/google/uuid v1.6.0
 	github.com/googleapis/gax-go/v2 v2.14.2
 	github.com/gorilla/websocket v1.5.4-0.20250319132907-e064f32e3674
-	github.com/grafana/pyroscope-go v1.2.3-0.20250624101038-0410c20e9693
+	github.com/grafana/pyroscope-go v1.2.4
 	github.com/gravitational/license v0.0.0-20250329001817-070456fa8ec1
 	github.com/gravitational/roundtrip v1.0.2
 	github.com/gravitational/teleport/api v0.0.0

--- a/integrations/terraform-mwi/go.mod
+++ b/integrations/terraform-mwi/go.mod
@@ -259,7 +259,7 @@ require (
 	github.com/gorilla/securecookie v1.1.2 // indirect
 	github.com/gorilla/websocket v1.5.4-0.20250319132907-e064f32e3674 // indirect
 	github.com/gosuri/uitable v0.0.4 // indirect
-	github.com/grafana/pyroscope-go v1.2.3-0.20250624101038-0410c20e9693 // indirect
+	github.com/grafana/pyroscope-go v1.2.4 // indirect
 	github.com/grafana/pyroscope-go/godeltaprof v0.1.8 // indirect
 	github.com/gravitational/license v0.0.0-20250329001817-070456fa8ec1 // indirect
 	github.com/gravitational/roundtrip v1.0.2 // indirect

--- a/integrations/terraform/go.mod
+++ b/integrations/terraform/go.mod
@@ -260,7 +260,7 @@ require (
 	github.com/gorilla/securecookie v1.1.2 // indirect
 	github.com/gorilla/websocket v1.5.4-0.20250319132907-e064f32e3674 // indirect
 	github.com/gosuri/uitable v0.0.4 // indirect
-	github.com/grafana/pyroscope-go v1.2.3-0.20250624101038-0410c20e9693 // indirect
+	github.com/grafana/pyroscope-go v1.2.4 // indirect
 	github.com/grafana/pyroscope-go/godeltaprof v0.1.8 // indirect
 	github.com/gravitational/license v0.0.0-20250329001817-070456fa8ec1 // indirect
 	github.com/gravitational/roundtrip v1.0.2 // indirect

